src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java

```diff
@@ -40,6 +40,35 @@ public interface ExamRepository extends JpaRepository<Exam, Long> {
             """)
     List<Exam> findByCourseIdWithExerciseGroupsAndExercises(@Param("courseId") long courseId);
 
+    // Find all exams for a single course (id) that are already visible and the user is registered or at least a tutor
+    @Query("""
+            SELECT DISTINCT ex
+            FROM Exam ex
+                LEFT JOIN ex.registeredUsers registeredUsers
+            WHERE ex.course.id = :courseId
+                AND ex.visibleDate <= :now
+                AND (registeredUsers.id = :userId
+                    OR ex.course.teachingAssistantGroupName IN :groupNames
+                    OR ex.course.editorGroupName IN :groupNames
+                    OR ex.course.instructorGroupName IN :groupNames)
+            """)
+    Set<Exam> findByCourseIdForUser(@Param("courseId") long courseId, @Param("userId") Long userId, @Param("groupNames") Set<String> groupNames, @Param("now") ZonedDateTime now);
+
+    // Find all exams for multiple courses (ids) that are already visible and the user is registered or at least a tutor
+    @Query("""
+            SELECT e
```

Comment:
Did you leave out ``DISTINCT`` here intentionally or does it not make sense at all? In the latter case, should we remove it in the first query above as well?